Sources/ApodiniUtils/Swift.swift

```diff
@@ -76,6 +81,45 @@ extension Sequence {
     public func dropNilValues<T>() -> [T] where Element == T? {
         self.compactMap { $0 }
     }
+    
+    /// Returns a set created by mapping the elements of the sequence using the specified block
+    public func mapIntoSet<Result: Hashable>(_ transform: (Element) throws -> Result) rethrows -> Set<Result> {
+        var retval = Set<Result>()
+        retval.reserveCapacity(self.underestimatedCount)
+        for element in self {
+            retval.insert(try transform(element))
+        }
+        return retval
+    }
+    
+    /// Returns a set created by compact-mapping the elements of the sequence using the specified block
+    public func compactMapIntoSet<Result: Hashable>(_ transform: (Element) throws -> Result?) rethrows -> Set<Result> {
+        var retval = Set<Result>()
+        retval.reserveCapacity(self.underestimatedCount / 2)
+        for element in self {
+            if let mapped = try transform(element) {
+                retval.insert(mapped)
+            }
+        }
+        return retval
+    }
+    
+    /// Returns the number of elements in the sequence that satisfy the predicate.
+    public func count(where predicate: (Element) -> Bool) -> Int {
+        var retval = 0
+        for element in self {
+            if predicate(element) {
+                retval += 1
+            }
+        }
+        return retval
+    }
+    
+    
+    /// Creates an array by consuming all of the sequence's elements.
+    public func intoArray() -> [Element] {
```

Comment:
i realize that this one is subjective, but IMO it is much nicer when working with chained operations on a sequence, and you can just call `.intoArray()` at the end, rather than having to wrap the whole function chaining expression in an `Array()` call. 

(it also makes more sense semantically, since the array construction happens at the end, but would otherwise appear at the very beginning of the expression)

Reply:
I Lukas's reasoning is very valid, though I don't like the method name at all, doesn't feel swifty to me 😋. Ideas I had were `.toArray()` or `.asArray()`, `.array()`, or even `.array`. I thought there was something about this in the Swift API Design Guidelines but I couldn't find it unfortunately.

Reply:
@moritzsternemann it should definitely be a function and not a computed property, because it's not an O(1) operation. As for the name, I chose `intoArray()` because it consumes the elements in the sequence and puts them into the array (it's not as extreme as in Rust, which is where I got the name from, but in Swift Sequences don't necessarily guarantee that you can iterate over them multiple times, so you might in fact end up with this call having side-effects (though it's incredibly unlikely). Hence the into.) If you want to rename it to `toArray()` that'd be absolutely fine with me, though I'd be against `asArray()` since that sounds like a type cast, which is not what's going on here.

Reply:
Makes sense, I'm fine with keeping `intoArray()` then 👍